### DISCUSSION TITLE
search house number by street name

### DISCUSSION THREAD

**Discussion by Exact1990** [komoot/photon#637] [Q&A]
Hello, we need some help.
We have problems with the results of adress searching. The geocoder does not give us options with streets even if we enter without any filtering parameters.

1 Why does auto-completion not work for streets? It seems to me that one of the functions of photon is: "search-as-you-type", and this should imply the offer of house numbers if there are not many street options found.
https://photon.komoot.io/api/?q=lenivka&limit=5&lat=55.7453267806082&lon=37.611316556002855
At the same time, if you specify the house number, it offers options.
https://photon.komoot.io/api/?q=lenivka%203&limit=5&lat=55.7453267806082&lon=37.611316556002855
2 Is it possible to make photon offer options with house numbers if only the street name is entered?

**Comment by lonvia** [comment]
Adding housenumbers to the suggestions is not something that is really feasible. There are a lot of houses on each street, which numbers should be shown? All of them? That's clearly not always possible unless you are happy with hundreds of suggestions. So just the ones closest to the point you have given? In your example Lenivka St is not exactly close to the point given. It's on the other side of the river. The one with important buildings? But what makes a number important then.

**Reply by Exact1990** [reply]
We want to get as many house numbers as we specify in the request (for example, in ascending order). 
For example, now we show only 5 results when entering "q =len" in the request , but we can get more if we specify limit = 10 or more

**Reply by lonvia** [reply]
You can experiment with lifting the [house number restriction here](https://github.com/komoot/photon/blob/742769de48c0dbc62270fd6f17ad904ce616f0b2/src/main/java/de/komoot/photon/query/PhotonQueryBuilder.java#L137) in the code.

**Reply by Exact1990** [reply]
Thank you
I will try to remember the experience with Java code
